lib/services/extensions-service-base.ts

```diff
@@ -0,0 +1,97 @@
+///<reference path="../.d.ts"/>
+"use strict";
+
+import path = require("path");
+import options = require("../common/options");
+import helpers = require("../helpers");
+
+export class ExtensionsServiceBase {
+	private extensionVersions: IStringDictionary = {};
+
+	constructor(public $fs: IFileSystem,
+				public $httpClient: Server.IHttpClient,
+				public $logger: ILogger) {
+		if (this.$fs.exists(this.versionsFile).wait()) {
```

Comment:
I'll implement this in another PR because we've already have too much "ENOENT" checks in our code.